core/src/datamatrix/DMECEncoder.cpp

```diff
@@ -121,8 +121,8 @@ static void CreateECCBlock(ByteArray& data, int codeOffset, int codeLength, int
 	for (int i = 0; i < codeLength; ++i) {
 		const auto m = ecc.back() ^ data[codeOffset + i * stride];
 		for (size_t k = ecc.size() - 1; k > 0; k--)
-			ecc[k] = ecc[k - 1] ^ mult(m, poly[k]);
-		ecc[0] = mult(m, poly[0]);
+			ecc[k] = ecc[k - 1] ^ mult(static_cast<uint8_t const>(m), poly[k]);
+		ecc[0] = mult(static_cast<uint8_t const>(m), poly[0]);
 	}
```

Comment:
east vs. west const is certainly an interesting debate. But worse than one or the other is mixing both. Please stick with the precedent set by the rest of the code.

Reply:
I agree, sorry not to have looked for const convention in use.

Reply:
Added commit c1b4ea1d5a11f5cd1a093d689dbea722d0fa29a3, but I will be still waiting opinions whether the whole `static_cast` thing is worth or not.